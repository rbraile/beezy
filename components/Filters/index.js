import React, { useState } from 'react'

const Filters = ({ handlerSearch }) => {
  const [search, setSearch] = useState('')

  const handleInputChange = e => {
    const { value } = e.currentTarget
    setSearch(value)
  }

  const handlerClick = (event, next) => {
    event.preventDefault()
    event.stopPropagation()
    if (search.trim().length >= 1) {
      next(search.trim())
    }
  }

  return (
    <div>
      <h2>Filters</h2>
      <form>
        <input
          type='text'
          name='search'
          value={search}
          onChange={handleInputChange}
        />
        <button type='submit' onClick={e => handlerClick(e, handlerSearch)}>
          Filter
        </button>
      </form>
    </div>
  )
}

export default Filters
