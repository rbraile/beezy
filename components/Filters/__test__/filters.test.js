import React from 'react'
import { mount, shallow } from 'enzyme'

import Filters from '../index'

describe('TEST = CHARACTERLIST', () => {
  test('TEST = FILTERS = SNAPSHOT', () => {
    const wrapper = mount(<Filters />)
    expect(wrapper).toMatchSnapshot()
  })

  test('TEST = FILTERS = Check has a 1 form', () => {
    const wrapper = mount(<Filters />)
    expect(wrapper.find('form')).toHaveLength(1)
  })
})
