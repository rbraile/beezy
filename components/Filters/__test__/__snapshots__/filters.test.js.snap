// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST = CHARACTERLIST TEST = FILTERS = SNAPSHOT 1`] = `
<Filters>
  <div>
    <h2>
      Filters
    </h2>
    <form>
      <input
        name="search"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        onClick={[Function]}
        type="submit"
      >
        Filter
      </button>
    </form>
    <button>
      Clear all
    </button>
  </div>
</Filters>
`;
